src/webvr-polyfill.js

```diff
@@ -13,46 +13,81 @@
  * limitations under the License.
  */
 
-var CardboardHMDVRDevice = require('./cardboard-hmd-vr-device.js');
-//var OrientationPositionSensorVRDevice = require('./orientation-position-sensor-vr-device.js');
-var FusionPositionSensorVRDevice = require('./fusion-position-sensor-vr-device.js');
+var CardboardVRDisplay = require('./cardboard-vr-display.js');
 var MouseKeyboardPositionSensorVRDevice = require('./mouse-keyboard-position-sensor-vr-device.js');
 // Uncomment to add positional tracking via webcam.
 //var WebcamPositionSensorVRDevice = require('./webcam-position-sensor-vr-device.js');
+var VRDisplay = require('./base.js').VRDisplay;
 var HMDVRDevice = require('./base.js').HMDVRDevice;
 var PositionSensorVRDevice = require('./base.js').PositionSensorVRDevice;
+var VRDisplayHMDDevice = require('./display-wrappers.js').VRDisplayHMDDevice;
+var VRDisplayPositionSensorDevice = require('./display-wrappers.js').VRDisplayPositionSensorDevice;
 
 function WebVRPolyfill() {
-  this.devices = [];
+  this.displays = [];
+  this.devices = []; // For deprecated objects
+  this.devicesPopulated = false;
+  this.nativeWebVRAvailable = this.isWebVRAvailable();
 
-  if (!this.isWebVRAvailable()) {
+  if (!this.nativeWebVRAvailable) {
     this.enablePolyfill();
   }
+  if (WebVRConfig.ENABLE_DEPRECATED_API && !this.isDeprecatedWebVRAvailable()) {
+    this.enableDeprecatedPolyfill();
+  }
 }
 
 WebVRPolyfill.prototype.isWebVRAvailable = function() {
-  return ('getVRDevices' in navigator) || ('mozGetVRDevices' in navigator);
+  return ('getVRDisplays' in navigator);
 };
 
+WebVRPolyfill.prototype.isDeprecatedWebVRAvailable = function() {
+  return ('getVRDevices' in navigator) || ('mozGetVRDevices' in navigator);
+};
 
-WebVRPolyfill.prototype.enablePolyfill = function() {
+WebVRPolyfill.prototype.populateDevices = function() {
+  if (this.devicesPopulated) {
+    return;
+  }
+  var vrDisplay = null;
   // Initialize our virtual VR devices.
   if (this.isCardboardCompatible()) {
-    this.devices.push(new CardboardHMDVRDevice());
+    var vrDisplay = new CardboardVRDisplay();
```

Comment:
variable already defined [above](#diff-7bfb91b61bb80cff41aba297525c4dc7R52). drop the `var`.